src/main/dashboard/app.tsx

```diff
@@ -127,59 +135,78 @@ function LoaderForm() {
+                  const file = files[0];
+                  setFile(file); // Update signal with the selected file
+
+                  // Read the file as text and store its content
+                  const reader = new FileReader();
+                  reader.onload = () => {
+                    setFileContent(reader.result as string); // Store the file content
+                  };
+                  reader.readAsText(file); // Read the file as text
+                } else {
+                  setFile(null); // Reset if no file is selected
+                  setFileContent(null); // Reset file content if no file is selected
+                }
+              }}
             />
           </Match>
         </Switch>
 
-        <button
-          class={styles.button}
-          disabled={shouldDisable()}
+        {/* Button Element */}
+        <button 
+          class={styles.button} 
+          disabled={shouldDisable()} 
           onClick={async () => {
-          switch (loaderType()) {
-            case 'URL':
-              setLoading(true);
-              await eureka.load(extensionURL());
-              setLoading(false);
-              break;
-            case 'Code':
-              setLoading(true);
-              await eureka.load(stringToDataURL(extensionCode()));
-              setLoading(false);
-              break;
-            case 'File':
-              if (extensionFile()) {
-                const reader = new FileReader();
-                reader.onload = async () => {
-                  setLoading(true);
-                  eureka.load(reader.result as string);
-                  setLoading(false);
-                };
-                reader.readAsText(extensionFile());
+            setLoading(true); // Set loading to true at the start of loading process
+            try {
+              switch (loaderType()) {
+                case 'URL':
+                  await eureka.load(extensionURL());
+                  break;
+                case 'Code':
+                  await eureka.load(stringToDataURL(extensionCode()));
+                  break;
+                case 'File':
+                  if (fileContent()) { // Use stored file content
+                    await eureka.load(stringToDataURL(fileContent()));
+                  }
+                  break;
               }
-              break;
-          }
-        }}>
+            
+            } catch (error) {
+              setErrorMessage("Error loading extension:" + error);
+              console.error("Error loading extension:", error); // Handle any errors that occur during loading
+            } finally {
+              setLoading(false); // Ensure loading is set to false after loading completes or fails
+            }
+          }}
+        >
           <FormattedMessage id="eureka.loader.load" default="Load Extension" />
         </button>
+        {errorMessage() && <p style={{ color: 'red' }}>{errorMessage()}</p>} {/* Display error message below the button */}
```

Comment:
`errorMessage` is a signal, use `errorMessage()` instead.
also, use className + css + Scratch's red

Reply:
I think I did. Also, please explain what you mean by `use className + css + Scratch's red`

Reply:
tsx
```typescript
<span className='errorText'>{errorMessage()}</span>
```

css file
```css
.errorText {
    font-size: 12px;
    color: var(--eureka-error-primary);
}
```

Reply:
I added it. Please let me know if i added the css in the wrong file. style.module.css

Reply:
Please check whether `var(--eureka-error-primary)` exists, and add that css variable in global.css if not

Reply:
Done!